src/OP.ts

```diff
@@ -96,6 +102,30 @@ export class OP {
     };
   }
 
+  public async newAuthenticationResponseWithSelected(
+    verifiedJwt: SIOP.VerifiedAuthenticationRequestWithJWT,
+    verifiableCredentials: VerifiableCredential[],
+    holderDID: string,
+    responseOpts?: {
+      nonce?: string;
+      state?: string;
+      // audience: string;
+      verification?: InternalVerification | ExternalVerification;
+    }
+  ): Promise<AuthenticationResponseWithJWT> {
+    const optionalPD = extractDataFromPath(verifiedJwt.payload, '$..presentation_definition');
```

Comment:
Can we not have the peAgent, or rather the user in the end come up with a list of VPs that we simply accept as an optional responseOpt?

Then the method to interact with to create the response stays the createJWTFromVerifiedRequest. We could make a small check there whether VPs are supplied when the request contained a PE definition.